hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsInputStream.java

```diff
@@ -145,6 +158,10 @@ public AbfsInputStream(
     this.readAheadQueueDepth = abfsInputStreamContext.getReadAheadQueueDepth();
     this.tolerateOobAppends = abfsInputStreamContext.isTolerateOobAppends();
     this.eTag = eTag;
+    this.fileStatusInformationPresent = StringUtils.isNotEmpty(eTag);
+    this.prefetchTriggerOnFirstRead =
+        abfsInputStreamContext.isPrefetchTriggerOnFirstRead()
+            && getFileStatusInformationPresent();
```

Comment:
Have removed the logic in the constructor. Now, prefetchTriggerOnFirstRead would just store the config value. It will be used in a method:
```
  private boolean effectiveReadAhead() {
    return (prefetchTriggerOnFirstRead && hasFileStatusInfo())
        || sequentialReadStarted;
  }
```

which shall be used to check if read ahead can be used or not in `readInternal`.